server/src/controllers/courseInstance.ts

```diff
@@ -17,45 +17,41 @@ import { HttpCode } from '../types/httpCode';
 import { idSchema } from '../types/general';
 import { Language } from '../types/language';
 import { CourseWithTranslation } from '../types/model';
-import { findCourseById } from './utils/course';
+import { findCourseById, findCourseWithTranslationById } from './utils/course';
 import { findUserById } from './utils/user';
 
-interface CourseInstanceWithCourseAndTranslation extends CourseInstance {
-  Course: CourseWithTranslation
-}
-
 export async function getCourseInstance(req: Request, res: Response): Promise<void> {
+  // Validate IDs.
+  const courseId: number = Number(req.params.courseId);
+  await idSchema.validate({ id: courseId });
+
   const instanceId: number = Number(req.params.instanceId);
   await idSchema.validate({ id: instanceId });
 
-  const instance: CourseInstanceWithCourseAndTranslation | null =
-      await models.CourseInstance.findByPk(
-        instanceId,
-        {
-          attributes: [
-            'id', 'sisuCourseInstanceId', 'gradingType', 'startingPeriod',
-            'endingPeriod', 'teachingMethod', 'minCredits', 'maxCredits',
-            'startDate', 'endDate', 'responsibleTeacher'
-          ],
-          include: {
-            model: Course,
-            attributes: ['id', 'courseCode'],
-            include: [
-              {
-                model: CourseTranslation,
-                attributes: ['language', 'courseName', 'department']
-              }
-            ]
-          }
-        }
-      ) as CourseInstanceWithCourseAndTranslation;
+  // Find course and course instance information.
+  const course: CourseWithTranslation = await findCourseWithTranslationById(
+    courseId, HttpCode.NotFound
+  );
 
+  const instance: CourseInstance | null = await CourseInstance.findByPk(
+    instanceId
+  );
+
+  // Verify that a course instance was found.
   if (!instance) {
     throw new ApiError(
       `course instance with ID ${instanceId} not found`, HttpCode.NotFound
     );
   }
 
+  // Verify that the course instance belongs to the found course.
+  if (instance.courseId != courseId) {
+    throw new ApiError(
+      `course instance with ID ${instanceId} belonging to course with ID ${courseId} not found`,
+      HttpCode.NotFound
```

Comment:
In PR #153 we use following message with http conflict when instance does not belong to the course.
```javascript
// Check that instance belongs to the course.
if (instance.courseId !== course.id) {
  throw new ApiError(
    `course instance with ID ${courseInstanceId} ` +
    `does not belong to the course with ID ${courseId}`,
    HttpCode.Conflict
  );
}
```

Should we use this here or update other PR to match this code?

Reply:
Tough to say. Ideally we wouldn't need to do this check at all, but I can change this to 409 for now.